### PR TITLE
#29 User authentication and authorisation

### PR DESCRIPTION
**Pull request by harerajo**
#### What does this PR do?

- Implements User authentication and authorization

#### Description of Task to be completed? 

- User should be able to log in, signup, and update the profile
- Add authorization to all relevant routes
- Added a few tests for user login and signup

#### How should this be manually tested?

Using postman, go to `localhost/5000/api`
- `post` method on `/auth/login` to login a user
- `post` method on `/auth/signup` to signup a user
- `put` method on `/auth/profile` to update user profile
- `get` method on `/auth/profile` to get user profile

#### What are the relevant Pivotal Tracker/Trello stories?
- https://trello.com/c/YRoCyHRd
- https://trello.com/c/NFoigxRq
- https://trello.com/c/0X9fHYEm

#### Screenshots (if appropriate)
-N/A

#### Questions:

- N/A

### CODE REVIEW

**Review by PrinceNiyonshuti — APPROVED** [review]
great work and large payload

**Review by thepatrickniyo — CHANGES_REQUESTED** [review]
Put `.env` file under `.gitignore` file. You don't need to push it to github

File: src/tests/e2e/auth.test.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+				done();
+			});
+	});
+	it("Should fail password did not match", (done) => {
+		chai
+			.request(app)
+			.post("/api/auth/login")
+			.send({ email: "david@gmail.com", password: "abcxdasdfgwert" })
+			.end((err, res) => {
+				console.log(res.body);
```
Comment (thepatrickniyo) [inline src/tests/e2e/auth.test.js:45]: Remove `console.log`. Don't have to push it to prouction

**Comment by harerajo** [comment]
> Put `.env` file under `.gitignore` file. You don't need to push it to github

is it allowed to push .env.example?

**Review by shemalucien — APPROVED** [review]
Good Work mn

**Pull request merged**